File: bindings/python/examples/async_invoke.py
"""The async client: invoke a recipe and await the durable result.

kx serve --journal /tmp/kx.db --content /tmp/kx-blobs --dev-allow-local
python examples/async_invoke.py
"""

from __future__ import annotations

import asyncio

from kortecx import AsyncKxClient


async def main() -> None:
    async with AsyncKxClient("http://127.0.0.1:50151") as kx:
        # wait_mode="events" reacts to the live event stream (lower latency than polling).
        result = await kx.invoke(
            "kx/recipes/echo", {"topic": "async hi"}, wait=True, wait_mode="events"
        )
        print("state", result.state, "instance", result.instance_id)


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/typescript/examples/async_invoke.ts
/**
 * Low-latency wait via the live event stream (`waitMode: "events"`).
 *
 * Identical result to the default poll wait, but reacts to the terminal Mote's
 * committed delta as it lands (sub-second) instead of polling the projection.
 *
 *   npx tsx examples/async_invoke.ts
 */

import { KxClient, type Result } from "@kortecx/sdk";

async function main(): Promise<void> {
  const kx = new KxClient("http://127.0.0.1:50151");
  try {
    const result = (await kx.invoke(
      "kx/recipes/echo",
      { topic: "low latency" },
      { wait: true, waitMode: "events" },
    )) as Result;
    console.log(result.ok ? "✓ committed" : `✗ ${result.state}`, result.instanceId);
    console.log("result:", result.text);
  } finally {
    kx.close();
  }
}

main().catch((err) => {
  console.error(String(err));
  process.exitCode = 1;
});
